nuxeo-csv-core/src/test/java/org/nuxeo/ecm/csv/core/TestCSVImporterCreateMode.java

```diff
@@ -468,24 +478,44 @@ public void shouldSetCreatorToTheUserImporting() throws InterruptedException {
 
         CSVImporterOptions options = new CSVImporterOptions.Builder().importMode(ImportMode.CREATE).build();
 
-        TransactionHelper.commitOrRollbackTransaction();
-        TransactionHelper.startTransaction();
+        txFeature.nextTransaction();
 
         try (CloseableCoreSession leelaSession = coreFeature.openCoreSession("leela")) {
 
+            // NXP-22172 : appears when leela does not have access to "/", yet ?
+            if (!leelaSession.exists(new PathRef("/"))) {
+                Thread.sleep(10 * 1000);
+                if (!leelaSession.exists(new PathRef("/"))) {
+                    log.error("NXP-22172: user session does not have access to '/' even after 10 seconds");
+                } else {
+                    log.error("NXP-22172: user session has access to '/' after 10 seconds");
+                }
+
+                // NXP-22172 : if the delay does not solve the issue, flush the leela session
+                leelaSession.save();
+                if (!leelaSession.exists(new PathRef("/"))) {
+                    log.error("NXP-22172: user session does not have access to '/' even after session flush");
+                    Assert.fail("NXP-22172: user session does not have access to '/'");
+                } else {
+                    log.error("NXP-22172: user session has access to '/' after 10 session flush");
+                }
+            }
```

Comment:
Seems rather heuristic no?
Calling `txFeature.nextTransaction()` instead of `session.save()` after `session.setACP` doesn't fix it?

Reply:
Totally heuristic and empirical, nobody can find a reason to the failing ..
After `session.setACP` both `session.save` and `txFeature.nextTransaction()` are called.